### ISSUE TITLE
Case insensitive [c] won't work in Russian

### ISSUE THREAD

**Issue by ITGypsy** [realm/realm-cocoa#2879]
[c] doesn't support Russian language
    
    func searchAutocompleteEntriesWithSubstring(substring: String) {
        suggestions.removeAll()
        
        let realmSuggestions = realm.objects(Cities).filter("ru BEGINSWITH[c] '\(substring)'")
        for suggestion in realmSuggestions {
            suggestions.append(suggestion.valueForKey("ru") as! String)
        }
        suggestionsTableView.reloadData()
    }
    
![ru_s](https://cloud.githubusercontent.com/assets/9496381/11267053/1094b50c-8eba-11e5-8790-f9dd8ba0e9e6.png) ![ru_b](https://cloud.githubusercontent.com/assets/9496381/11267059/14b0ac04-8eba-11e5-93d6-db42357d38ef.png)

In English it works as expected
    
    func searchAutocompleteEntriesWithSubstring(substring: String) {
        suggestions.removeAll()
        
        let realmSuggestions = realm.objects(Cities).filter("en BEGINSWITH[c] '\(substring)'")
        for suggestion in realmSuggestions {
            suggestions.append(suggestion.valueForKey("en") as! String)
        }
        suggestionsTableView.reloadData()
    }
    
![en_c](https://cloud.githubusercontent.com/assets/9496381/11267075/38360782-8eba-11e5-96a3-9d42e08bdd7b.png)

Please fix.

Realm 0.96.2 / Xcode 7.1 / OSX 10.11.1
You can [find me](http://www.facebook.com/arthur.kutarba) on Facebook for further assistance.

**Comment by jpsim** [comment]
Realm only currently supports English/US for sorting (#2004) and case insensitive searches (now tracked by this issue).

Until we add support for this, you could store both your normal string along with a "normalized" version in your data model, where you perform the case folding before storing the value in Realm, perhaps using [CFStringTransform](http://nshipster.com/cfstringtransform/). You could then search on that string (also normalizing your search input).

**Comment by Rusik** [comment]
Please add case insensitive searches for Cyrillic :(

**Comment by jpsim** [comment]
@Rusik that's exactly what this ticket is tracking

**Comment by Rusya13** [comment]
+

**Comment by ITGypsy** [comment]
It's been a while. What's the status?

**Comment by mixdesign** [comment]
+1. Must have

**Comment by Galeas** [comment]
@ITGypsy nobody cares, I guess :(

**Comment by mixdesign** [comment]
@ITGypsy @Galeas any alternatives to realm, except core data?

**Comment by jpsim** [comment]
We certainly care about adding this feature, but have to weigh it against all the other work that needs to be done.

The fact that it's easy to work around for most cases (see recommendations in https://github.com/realm/realm-cocoa/issues/2879#issuecomment-158101749) is one reason why this isn't prioritized more highly.

**Comment by ITGypsy** [comment]
Yes indeed, it's fairly easy to work around. Just use `.lowercaseString` when saving a copy of your string to your model, as well as when searching.

```
if realmSuggestions.filter("lowercaseRU BEGINSWITH '\(substring)'").count > 0 {
                
        for suggestion in realmSuggestions {
                    
            suggestions.append("\(suggestion.valueForKey("ru") as! String)")
        }

        suggestionsTableView.reloadData()
}
```


But still, please don't keep putting it off. A work-around is a work-around, not a solution.

**Comment by NosovPavel** [comment]
@jpsim hi! Can you tell me when are you going to add this feature? There is no way to add normalized string to each Object because of it takes a lot of time and this is not the right way.

**Comment by jpsim** [comment]
No one's working on this at the moment and I have no estimate of when it'll be available.

**Comment by APUtils** [comment]
Case insensitive search works with Russian language if you use it with `cd`:
```
.filter("name CONTAINS[cd] %@", ...)
```

**Comment by Anakros** [comment]
Yeah, it works for case-insensitive search for me with `CONTAINS[cd]`, but not `CONTAINS[c]` or `LIKE[cd] *word*` or `LIKE[c] *word*`. It's pretty strange.

**Comment by dimlksin2** [comment]
as relevant, is there a solution?

**Comment by tiu93** [comment]
```CONTAINS[cd]``` Doesn't work for me.
